code/controllers/subsystem/familytree.dm

```diff
@@ -156,7 +156,8 @@ SUBSYSTEM_DEF(familytree)
 	if(!H || !status || istype(H, /mob/living/carbon/human/dummy))
 		return
 	//Exclude princes and princesses from having their parentage calculated.
-	if(H.job in excluded_jobs)
+	var/job_check = H.mind?.assigned_role?.parent_job ? H.mind.assigned_role.parent_job.title : H.job
+	if(job_check in excluded_jobs)
```

Comment:
make this a proc, a global one is fine passing in the mob and exluded list. 

Or... you could make this a list of job types like I suggested

Reply:
checking both is ok

Reply:
You either have to double up the base "parent" of the advclass and the class parent or check two different things anyway

```
var/list/jobs = list(
	/datum/job/x
	/datum/job/advclass/x,
)
```

`if(is_type_in_list(h.mind.assigned_role, jobs)`
 or 
`if(is_type_in_list(SSjob.GetJob(h.job), jobs)`

Reply:
i did it a bit differently, the strings were turned into datum and now it checks directly for the type and see if it is the same.